### ISSUE TITLE
Too many edges

### ISSUE THREAD

**Issue by tygrysio** [pgRouting/osm2pgrouting#297]
Fedora 34. pgRouting-3.2.1. osm2pgrouting-2.3.8

When it imports the data for that area.
https://www.openstreetmap.org/#map=19/53.43298/14.54038
It gets something like this.

![placSzarychSzeregów](https://user-images.githubusercontent.com/22832623/135059792-7e426a80-0f9e-4aa0-8dfb-071dbc3bff60.png)

<osm version="0.6" generator="CGImap 0.8.5 (282149 spike-07.openstreetmap.org)" copyright="OpenStreetMap and contributors" attribution="http://www.openstreetmap.org/copyright" license="http://opendatacommons.org/licenses/odbl/1-0/">

`<osm version="0.6" generator="CGImap 0.8.5 (282149 spike-07.openstreetmap.org)" copyright="OpenStreetMap and contributors" attribution="http://www.openstreetmap.org/copyright" license="http://opendatacommons.org/licenses/odbl/1-0/">
<way id="829873237" visible="true" version="1" changeset="88476252" timestamp="2020-07-24T21:22:59Z" user="diverpl" uid="603523">
<nd ref="1648910558"/>
<nd ref="7152402421"/>
<nd ref="7152402418"/>
<nd ref="7152402419"/>
<nd ref="7152402417"/>
<nd ref="7152402420"/>
<nd ref="7152402414"/>
<nd ref="7152402416"/>
<nd ref="7152402415"/>
<nd ref="4519628705"/>
<nd ref="7152402413"/>
<nd ref="1648910477"/>
<nd ref="7152402412"/>
<nd ref="4519628707"/>
<nd ref="7152402411"/>
<nd ref="4519628706"/>
<nd ref="7152402409"/>
<nd ref="1648910450"/>
<nd ref="7152402410"/>
<nd ref="4519628704"/>
<nd ref="7152402408"/>
<nd ref="4519628703"/>
<nd ref="7152402407"/>
<nd ref="1648910441"/>
<nd ref="7152402405"/>
<nd ref="4519628702"/>
<nd ref="7152402406"/>
<nd ref="4519628701"/>
<nd ref="7152402404"/>
<nd ref="4519628700"/>
<nd ref="7152402403"/>
<nd ref="1648910429"/>
<nd ref="7152402402"/>
<nd ref="4519628699"/>
<nd ref="1648910398"/>
<nd ref="4519628694"/>
<nd ref="4519628698"/>
<nd ref="4519628697"/>
<nd ref="1648910392"/>
<nd ref="4519628695"/>
<nd ref="7152402401"/>
<nd ref="4519628696"/>
<nd ref="7152402396"/>
<nd ref="7152402397"/>
<nd ref="7152402395"/>
<nd ref="7152402398"/>
<nd ref="7152402394"/>
<nd ref="7152402399"/>
<nd ref="7152402393"/>
<nd ref="7152402400"/>
<nd ref="1648910389"/>
<tag k="access" v="no"/>
<tag k="bus" v="yes"/>
<tag k="highway" v="service"/>
<tag k="oneway" v="yes"/>
</way>
</osm>`

**Comment by tygrysio** [comment]
Yes. Circles are nodes in the ways_vertices_pgr table. I used the map image to see it better.

**Comment by cvvergara** [comment]
Roads are segmented like with speed bumps, stop signs, pedestrian crossings, etc

**Comment by cvvergara** [comment]
```
osm2pgrouting -f issue297.osm -c "/usr/local/share/osm2pgrouting/mapconfig.xml" -d city_routing -U user -W user --clean
```

<img width="814" height="362" alt="Image" src="https://github.com/user-attachments/assets/7d553adb-a612-45d5-b65f-140048adc73e" />

I get a lot of  line breaks for the same osm_id.

You can use [pgr_contractionLinear](https://docs.pgrouting.org/latest/en/pgr_contractionLinear.html) to simplify the graph

But it does what it needs to do, create a routing graph
